src/main/java/jdbchomework/dao/jdbc/ProjectsJdbcDao.java

```diff
@@ -2,11 +2,66 @@
 
 import jdbchomework.dao.model.ProjectsDao;
 import jdbchomework.entity.Project;
-import org.hibernate.SessionFactory;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 
 public class ProjectsJdbcDao extends AbstractDao<Project> implements ProjectsDao {
 
-    public ProjectsJdbcDao(SessionFactory sessionFactory, String entity, Class<Project> clazz) {
-        super(sessionFactory, entity, clazz);
+    private static org.slf4j.Logger log = LoggerFactory.getLogger(ProjectsJdbcDao.class);
+
+    public ProjectsJdbcDao(Connection connection, String table, String column) {
+        super(connection, table, column);
+    }
+
+    @Override
+    public void add(Project toAdd) {
+        String name = toAdd.getName();
+        int cost = toAdd.getCost();
+        try (PreparedStatement statement = connection
+                .prepareStatement("INSERT INTO projects (name, cost) VALUES (?,?)")) {
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            statement.setString(1, name);
+            statement.setInt(2, cost);
+            statement.executeUpdate();
+            System.out.println(toAdd.getName() + " successfully added to DB");
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Cannot connect to DB" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)